src/pal/src/CMakeLists.txt

```diff
@@ -6,6 +6,14 @@ project(coreclrpal)
 
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
+if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    # Enable @rpath support for shared libraries.
+    set(MACOSX_RPATH ON)
+    if(CMAKE_VERSION VERSION_EQUAL 3.0 OR CMAKE_VERSION VERSION_GREATER 3.0)
+        cmake_policy(SET CMP0042 NEW)
```

Comment:
Based on my reading of [the documentation](http://www.cmake.org/cmake/help/v3.2/policy/CMP0042.html) we will get a warning from CMake if this policy is not set to NEW.  If we depended on a newer version of CMake (via cmake_minimum_required, then it would automatically be set).  I will keep this as is.